packages/ui/src/theme/createTheme/createComponentCSS.ts

```diff
@@ -0,0 +1,152 @@
+import kebabCase from 'lodash/kebabCase.js';
+import { DefaultTheme, WebTheme } from '../types';
+import { cssValue, isDesignToken } from './utils';
+import { ComponentsTheme } from '../components';
+import { isFunction, splitObject } from '../../utils';
+import { BaseTheme, CSSProperties } from '../components/utils';
+
+/**
+ * This will take an object like:
+ * {paddingTop:'20px',color:'{colors.font.primary}'}
+ * and turn it into a CSS string:
+ * `padding-top:20px; color: var(--colors-font-primary);`
+ */
+export function propsToString(props: CSSProperties): string {
+  return Object.entries(props)
+    .map(([key, value]) => {
+      const _value = isDesignToken(value)
+        ? value.toString()
+        : // @ts-ignore
+          cssValue({ value });
+      return `${kebabCase(key)}:${_value}; `;
```

Comment:
Nit: Would it make sense to move this usage of `kebabCase()` into the utils file which is already importing it for its use in `cssValue()`? That would contain all non-test usage of `kebabCase` in one place and allow us to remove its import here.

Reply:
Nice change here, better than what I had in mind 👍